src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.TestPlatform.Build.Utils
+{
+    public static class ArgumentEscaper
+    {
+        /// <summary>
+        /// Undo the processing which took place to create string[] args in Main,
+        /// so that the next process will receive the same string[] args
+        /// 
+        /// See here for more info:
+        /// http://blogs.msdn.com/b/twistylittlepassagesallalike/archive/2011/04/23/everyone-quotes-arguments-the-wrong-way.aspx
+        /// </summary>
+        /// <param name="arg"></param>
+        /// <returns>Return original string passed by client</returns>
+        public static string EscapeArgForProcessStart(string arg)
```

Comment:
I renamed it to HandleEscapeSequenceInArgForProcessStart. This method make changes in args to pass Process.Start. We we have to pass this to cmd then args will get handle in some other way.